src/vm/ceemain.cpp

```diff
@@ -956,6 +952,10 @@ void EEStartupHelper(COINITIEE fFlags)
         // throws on error
         SetupThread();
 
+        // This isn't done as part of InitializeGarbageCollector() above because thread
+        // creation requires AppDomains to have been set up.
+        FinalizerThread::FinalizerThreadCreate();
+
```

Comment:
How about we move this down to line 1011? On desktop there is a function InitializeGarbageCollection that contains the code:

    FinalizerThread::FinalizerThreadCreate();
    SetGarbageCollectorFullyInitialized();

It appears that function was refactored as part of the local GC work, but I would keep the ordering as similar as possible unless you've already tested that option and found it caused a problem.

Reply:
@Maoni0 @swgillespie - Adding the GC folks in case they want to weigh in any further.

Reply:
seems better to me.

Reply:
looks fine to me.